src/extension/experimentService/experimentService.ts

```diff
@@ -0,0 +1,130 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for details.
+
+import * as Configstore from "configstore";
+import * as https from "https";
+import { PreviewVersionPromotion} from "./experiments/previewVersionPromotion";
+import { TelemetryHelper } from "../../common/telemetryHelper";
+
+export enum ExperimentStatuses {
+    FAILURE = "failure",
+    SUCCESS = "success",
+    SKIPPED = "skipped",
+}
+
+export interface ExperimentConfig {
+    experimentName: string;
+    popCoveragePercent: number;
+    enabled: boolean;
+}
+
+export interface ExperimentParameters extends ExperimentConfig {
+    [key: string]: any;
+    extensionId?: string;
+}
+
+export interface ExperimentResult {
+    resultStatus: ExperimentStatuses;
+    updatedExperimentParameters: ExperimentParameters;
+}
+
+export class ExperimentService {
+    private readonly endpointURL: string;
+    private readonly configName: string;
+    private readonly availableExperiments: any;
+    private config: Configstore;
+    private downloadedExperimentsConfig: Array<ExperimentConfig> | null;
+    private downloadConfigProm: Promise<ExperimentConfig[]> | null;
```

Comment:
Maybe rename this to `downloadConfigReq` or `downloadConfigRequest`. `Prom` ending looks not very obvious